gitops/gitops.go

```diff
@@ -17,190 +17,20 @@ package gitops
 
 import (
 	"fmt"
-	"os/exec"
 	"path/filepath"
 
 	appstudiov1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
 	"github.com/redhat-appstudio/application-service/gitops/prepare"
-	appstudioshared "github.com/redhat-appstudio/managed-gitops/appstudio-shared/apis/appstudio.redhat.com/v1alpha1"
 	"github.com/spf13/afero"
 )
 
-type Executor interface {
-	Execute(baseDir, command string, args ...string) ([]byte, error)
-	GenerateParentKustomize(fs afero.Afero, gitOpsFolder string) error
-}
-
-// GenerateAndPush takes in the following args and generates the gitops resources for a given component
-// 1. outputPath: Where to output the gitops resources to
-// 2. remote: A string of the form https://$token@github.com/<org>/<repo>. Corresponds to the component's gitops repository
-// 2. component: A component struct corresponding to a single Component in an Application in AS
-// 4. The executor to use to execute the git commands (either gitops.executor or gitops.mockExecutor)
-// 5. The filesystem object used to create (either ioutils.NewFilesystem() or ioutils.NewMemoryFilesystem())
-// 6. The branch to push to
-// 7. The path within the repository to generate the resources in
-// 8. The gitops config containing the build bundle;
-// Adapted from https://github.com/redhat-developer/kam/blob/master/pkg/pipelines/utils.go#L79
-func GenerateAndPush(outputPath string, remote string, component appstudiov1alpha1.Component, e Executor, appFs afero.Afero, branch string, context string, gitopsConfig prepare.GitopsConfig) error {
+func GenerateTektonBuild(outputPath string, component appstudiov1alpha1.Component, appFs afero.Afero, context string, gitopsConfig prepare.GitopsConfig) error {
```

Comment:
I think this wrapper and generate.go func can be merged and reduced further, nothing special happening here IMO

Reply:
I've removed gitops.go and gitops_test.go and kept generate.go and generate_test.go.  It calls generateBuild, so the tests in generate_test.go are based on the similar tests from TestGenerateBuild in generate_build_test.go.

Note, that you simply have to change line 33: `fs := ioutils.NewMemoryFilesystem()` to `fs := ioutils.NewFilesystem()` and provide the local file path at line 32, to test and see the generated files locally.

Also, the two tests are run 'independent' of each since the code generation is done in their own `testFolder`.  (test1 and test2, in this case). I think some of the existing tests generate over the same output folder.

Reply:
Also, in the original description of this PR, I mentioned we could potentially get rid of generate.go and put everything in generate_build.go.  Let me know if you want to do this.  (On the other hand, the stuff in generate.go, is basically a wrapper, so perhaps you want to keep this separate).